cassiopeia/solver/VanillaGreedySolver.py

```diff
@@ -5,65 +5,118 @@
         self.fuzzy_solver = fuzzy_solver
 
-    def solve(self):
-        """Implements the solver routine for Cassiopeia-Greedy
-
-        Cassiopeia Greedy proceeds by identifying first a mutation that is most
-        likely to have occurred earliest in the phylogeny, partitioning samples
-        into two sets based on the presence or absence of this mutation, and then
-        recursing on these two sets until sets of one sample are obtained.
-        To note, at each partitioning step, samples with missing data at the
-        character of interest are assigned to one of the two partitions based on
-        the function stored in self.missing_data_classifier.
-        """
-        pass
-
-    def find_split(self) -> Tuple[int, str]:
-        """Identifies a (character, state) pair on which to split.
-
-        Identifies the most (character, state) pair that was most likely to
-        have occurred earliest in the phylogeny. Without priors, this is simply
-        the most frequently occurring mutation. With priors, this is is the most
-        frequently occurring rare mutation (i.e. frequencies are weighted by a
-        prior probability of the mutation occurring).
-        """
-        pass
-
     def perform_split(
-        self, samples: List[int], split: Tuple[int, str]
+        self, samples: List[int] = None
     ) -> Tuple[List[int], List[int]]:
-        """Performs a partition based on the (character, state) pair.
+        """Performs a partition based on the most frequent (character, state) pair.
         
         Uses the (character, state) pair to split the list of samples into
         two partitions. In doing so, the procedure makes use of the missing
         data classifier passed into the class.
 
         Args:
             samples: A list of samples to partition
-            split: A (character, state) tuple to use as the character-split
         
         Returns:
             A tuple of lists, representing the left and right partitions
         """
-        pass
+        if not samples:
+            samples = list(range(self.prune_cm.shape[0]))
+        F = utils.compute_mutation_frequencies(samples)
+        freq = 0
+        char = 0
+        state = 0
+        for i in F:
+            for j in F[i]:
+                if j != self.missing_char and j != '0':
+                    if F[i][j] > freq and F[i][j] < len(samples) - F[i][self.missing_char]:
+                        char, state = i,j
+                        freq = F[i][j]
+        if freq == 0:
+            return self.random_nontrivial_cut(samples)
+        S = []
+        Sc = []
+        missing = []
+
+        for i in samples:
+            if self.prune_cm.iloc[i, char] == state:
+                S.append(i)
+            elif self.prune_cm.iloc[i, char] == -1:
+                missing.append(i)
+            else:
+                Sc.append(i)
+        
+        S, Sc = self.assign_missing_average(S, Sc, missing)
```

Comment:
A user should be able to pass a `missing_data_classifier` function. So let's not call `self.assign_missing_average` here, but rather `self.missing_data_classifier`. 

We can have a separate library for `greedy_missing_data_classifiers.py`